cnvlib/call.py

```diff
@@ -148,15 +148,9 @@ def absolute_threshold(cnarr, ploidy, thresholds, is_reference_male):
 
 def absolute_clonal(cnarr, ploidy, purity, is_reference_male, is_sample_female):
     """Calculate absolute copy number values from segment or bin log2 ratios."""
-    absolutes = np.zeros(len(cnarr), dtype=np.float_)
-    for i, row in enumerate(cnarr):
-        # TODO by_chromosome to reduce number of calls to this
-        ref_copies, expect_copies = _reference_expect_copies(
-            row.chromosome, ploidy, is_sample_female, is_reference_male)
-        absolutes[i] = _log2_ratio_to_absolute(
-            row.log2, ref_copies, expect_copies, purity)
-    return absolutes
+    df = absolute_dataframe(cnarr, ploidy, purity, is_reference_male, is_sample_female)
```

Comment:
`absolute_clonal()` uses a subset of the results from `absolute_dataframe()` so I figured this could be refactored here.